crates/subspace-networking/src/behavior/custom_record_store.rs

```diff
@@ -381,8 +392,64 @@ impl<'a> RecordStorage<'a> for ParityDbRecordStorage {
         self.save_data(key, None);
     }
 
-    // TODO: add BTree iterator
     fn records(&'a self) -> Self::RecordsIter {
-        Vec::new().into_iter()
+        let rec_iter_result: Result<ParityDbRecordIterator, parity_db::Error> = try {
+            let btree_iter = self.db.iter(self.column_id)?;
+            ParityDbRecordIterator::new(btree_iter)?
+        };
+
+        match rec_iter_result {
+            Ok(rec_iter) => rec_iter,
+            Err(err) => {
+                error!(?err, "Can't create Parity DB record storage iterator.");
+
+                ParityDbRecordIterator::empty()
```

Comment:
Why returning empty iterator instead of an error? Silencing these doesn't seem nice.

Reply:
https://github.com/libp2p/rust-libp2p/issues/3035